src/gval/utils/loading_datasets.py

```diff
@@ -320,3 +227,230 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
     else:
         return xr_object
+
+
+def _create_circle_mask(
+    sizes: int | Tuple[int], center: Tuple[Number, Number], radius: Number
+) -> np.ndarray:
+    """
+    Function to create a circle mask
+
+    Parameters
+    ----------
+    sizes : Int or Tuple of Int
+        Size of xarray's in number of pixels or tuple with the x and y sizes, respectively.
+    center : Tuple of Number
+        Tuple with the center coordinates (x, y).
+    radius : Number
+        Radius of the circle.
+
+    Returns
+    -------
+    mask : np.ndarray
+        Numpy array with the circle mask.
+    """
+    Y, X = np.ogrid[: sizes[1], : sizes[0]]
+    dist_from_center = np.sqrt((X - center[0]) ** 2 + (Y - center[1]) ** 2)
+    mask = dist_from_center <= radius
+    return mask
+
+
+def _create_xarray(
+    upper_left: Tuple[Number, Number],
+    lower_right: Tuple[Number, Number],
+    sizes: int | Tuple[int, int],
+    band_params: Iterable[Tuple[Number, Number, Tuple[Number, Number], Number]],
+    nodata_value: Optional[Number] = None,
+    encoded_nodata_value: Optional[Number] = None,
+    shapes: str = "circle",
+    band_dim_name: str = "band",
+    return_dataset: bool = False,
+) -> xr.DataArray | xr.Dataset:
+    """
+    Function to create xarray's with circles per band, and optionally return a Dataset
+
+    Parameters
+    ----------
+    upper_left : Tuple of Number
+        Tuple with the upper left coordinates (x, y).
+    lower_right : Tuple of Number
+        Tuple with the lower right coordinates (x, y).
+    sizes : Int or Tuple of Int
+        Size of xarray's in number of pixels or tuple with the x and y sizes, respectively.
+    band_params : Iterable of Tuples
+        Each tuple represents the parameters of each band comprised of:
+        (background_value, circle_value, circle_center, circle_radius).
+
+        Circle center is a tuple with (x, y) coordinates.
+
+        The data types are:
+            background_value: Number
+            circle_value: Number
+            circle_center: tuple of Number
+            circle_radius: Number
+    nodata_value : Optional[Number], default = None
+        Nodata value to use for the xarray's.
+    encoded_nodata_value : Optional[Number], default = None
+        Nodata value to use for the encoded xarray's.
+    shapes : str, default = 'circle'
+        Shape of mask within the xarray's. Currently only 'circle' is supported.
+    band_dim_name : str, default = 'band'
+        Name of the band dimension.
+    return_dataset : bool, default = False
+        Whether to return a Dataset instead of a DataArray.
+
+    Raises
+    ------
+    ValueError
+        If the shape is not supported.
+
+    Returns
+    -------
+    data_array : xarray.DataArray or xarray.Dataset
+        Xarray with the circles per band.
+    """
+
+    # Handle sizes
+    if isinstance(sizes, Number):
+        sizes = int(sizes)
+        sizes = (sizes, sizes)
+
+    # handle shapes
+    if shapes == "circle":
```

Comment:
I would remove 318-325  and delegate check for valid shape to the loop starting 330 unless this is for memory purposes or some future masks may generalize to multiple shapes.

Reply:
Yes, the purpose of this is to support other array creation functions eventually.